From: Sameeh Jubran <sameehj@amazon.com>
Date: Sun, 6 Oct 2019 15:33:25 +0300
Subject: net: ena: ethtool: get_channels: use combined only
Patch-mainline: v5.5-rc1
Git-commit: 9a037b06137b46bea6ed2fade2537cb820c1fd7c
References: bsc#1154492

Since we use the same IRQ and NAPI to service RX and TX then we need to
use a combined channel instead of rx and tx channels.

Signed-off-by: Sameeh Jubran <sameehj@amazon.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/net/ethernet/amazon/ena/ena_ethtool.c |   10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

--- a/drivers/net/ethernet/amazon/ena/ena_ethtool.c
+++ b/drivers/net/ethernet/amazon/ena/ena_ethtool.c
@@ -734,14 +734,8 @@ static void ena_get_channels(struct net_
 {
 	struct ena_adapter *adapter = netdev_priv(netdev);
 
-	channels->max_rx = adapter->num_io_queues;
-	channels->max_tx = adapter->num_io_queues;
-	channels->max_other = 0;
-	channels->max_combined = 0;
-	channels->rx_count = adapter->num_io_queues;
-	channels->tx_count = adapter->num_io_queues;
-	channels->other_count = 0;
-	channels->combined_count = 0;
+	channels->max_combined = adapter->num_io_queues;
+	channels->combined_count = adapter->num_io_queues;
 }
 
 static int ena_get_tunable(struct net_device *netdev,
